spec/controllers/groups_controller_spec.rb

```diff
@@ -241,6 +241,53 @@
       end
     end
 
+    describe '#create_groups_when_students_work_alone' do
+      before :each do
```

Comment:
You have the right idea here, but should use `let` and `let!` instead of `before :each`. See [here](https://www.codewithjason.com/difference-let-let-instance-variables-rspec/), for example.

Also, you shouldn't hard-code specific user names. Instead, use `create_list` to create two students, and then compute the student's usernames down below for `data`.